Sources/MapboxMaps/Style/Style+Localization.swift

```diff
@@ -40,20 +51,33 @@ extension Style {
 
     /// Filters through source to determine supported locale styles.
     /// This is needed for v7 support
-    internal func getLocaleValue(locale: Locale) -> String {
+    internal func getLocaleValue(locale: Locale) -> String? {
+        // Reference for supported languages: https://docs.mapbox.com/ios/navigation/guides/localization-and-internationalization/#supported-languages
+        let supportedLocaleIdentifiers = ["ar", "zh", "en", "fr", "de", "it", "ja", "ko", "pt", "ru", "zh-Hans", "es", "zh-Hant", "vi"]
+
+        // Do nothing if we do not support the locale
+        if !supportedLocaleIdentifiers.contains(locale.languageCode!) {
+            return nil
+        }
+
         let vectorSources = allSourceIdentifiers.filter { source in
             return source.type == .vector
         }
 
         for sourceInfo in vectorSources {
             do {
-                let tempSource = try source(withId: sourceInfo.id) as VectorSource
+                if locale.identifier.starts(with: "zh") {
+                    let tempSource = try source(withId: sourceInfo.id) as VectorSource
 
-                if tempSource.url?.contains("mapbox.mapbox-streets-v7") == true {
-                    if locale.identifier.contains("zh") {
+                    if tempSource.url?.contains("mapbox.mapbox-streets-v7") == true {
                         // v7 styles does not support value of "name_zh-Hant"
-                        if locale.identifier == SupportedLanguage.traditionalChinese.rawValue {
-                            return SupportedLanguage.chinese.rawValue
+                        if locale.identifier == "zh-Hant" {
+                            return "zh"
+                        }
+                    } else if tempSource.url?.contains("mapbox.mapbox-streets-v8") == true {
+                        // Return traditional chinese if the Locale is Taiwan
+                        if locale.identifier == "zh-Hant-TW" {
+                            return "zh-Hant"
                         }
```

Comment:
Why is TW the only region code that we need to strip?

Reply:
Do we support other region codes?

Reply:
not as of now